File: tutorials/testing/c++/graph_alist_empty.cpp
#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "GraphAdjList.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// create a graph
	GraphAdjList<string, string, int> graph;

	string kevin_bacon = "Kevin_Bacon_(I)", denzel_washington = "Denzel_Washington", noone= "NoOne";

	graph.addVertex(kevin_bacon, "");
	graph.addVertex(denzel_washington, "");
	graph.addVertex(noone, "");

	graph.addEdge(kevin_bacon, denzel_washington, 1);

	


	for (auto& edge : graph.outgoingEdgeSetOf(kevin_bacon)) {
		string from = edge.from(), to = edge.to();
		if (to != "Denzel_Washington")
			graph.getVisualizer(to)->setColor("turquoise");
	}
	for (auto& edge : graph.outgoingEdgeSetOf(denzel_washington)) {
		string from = edge.from(), to = edge.to();
		if (to != kevin_bacon)
			graph.getVisualizer(to)->setColor("orange");
	}
	for (auto& edge : graph.outgoingEdgeSetOf(noone)) {
		string from = edge.from(), to = edge.to();
		if (to != kevin_bacon)
			graph.getVisualizer(to)->setColor("orange");
	}

	
	
	return 0;
}

File: tutorials/testing/python/graph_alist_empty.py
from bridges.graph_adj_list import *
from bridges.bridges import *

import sys


def main():

    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    
    # creating a 3 vertices graph with one edge.
    g = GraphAdjList()
    a1 = "Kevin_Bacon_(I)"
    a2 = "Denzel_Washington"
    a3 = "Noone"
    g.add_vertex(a1, "")
    g.add_vertex(a2, "")
    g.add_vertex(a3, "")
    g.add_edge(a1, a2)
    
    # works
    for edge in g.out_going_edge_set_of(a1):
        f = edge.fromv
        to = edge.tov
    
    # does not work
    for edge in g.out_going_edge_set_of(a2):
        f = edge.fromv
        to = edge.tov

    # does not work
    for edge in g.out_going_edge_set_of(a3):
        f = edge.fromv
        to = edge.tov
            
if __name__ == "__main__":
    main()

